server/expire_cache.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"sync"
+	"time"
+)
+
+type cacheItem struct {
+	key    uint64
+	value  interface{}
+	expire time.Time
+}
+
+// ExpireRegionCache is an expired region cache.
+type ExpireRegionCache struct {
+	sync.RWMutex
+
+	items      map[uint64]cacheItem
+	ttl        time.Duration
+	gcInterval time.Duration
+}
+
+// NewExpireRegionCache returns a new expired region cache.
+func NewExpireRegionCache(gcInterval time.Duration, ttl time.Duration) *ExpireRegionCache {
+	c := &ExpireRegionCache{
```

Comment:
Yep, but now we only have one package, and the style is not unified, so it is better to do this in a new PR.